src/core/load-fragment.c

```diff
@@ -3830,7 +3830,13 @@ static int load_from_path(Unit *u, const char *path) {
                         if (r >= 0)
                                 break;
                         filename = mfree(filename);
-                        if (r != -ENOENT)
+
+                        /* ENOENT means that the file is missing or is a dangling symlink.
+                         * ENOTDIR means that one of paths we expect to be is a directory
+                         * is not a directory, we should just ignore that.
+                         * EACCES means that the directory or file permissions are wrong.
+                         */
+                        if (!IN_SET(r, -ENOENT, -ENOTDIR, -EACCES))
```

Comment:
shouldn't we at least log about the EACCES thing at debug level?

(ENOENT/ENOTDIR doesn#t need debug logging, I think)